docassemble/ALToolbox/business_days.py

```diff
@@ -0,0 +1,87 @@
+import holidays  
+import pandas as pd
+import datetime 
+from datetime import date as dt  
+from docassemble.base.util import as_datetime
+"""
+  External docs: 
+  1. https://github.com/dr-prodigy/python-holidays
+  2. https://pypi.org/project/holidays/
+  3. https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html
+  Note:
+  There seems to be a core issue in holidays module, where if you delete a holiday cross the year, it will add in holidays from the previous year in error. For example, 
+      12/31/2021: New Year’s Day (Observed) and 01/01/2022: New Year’s Day 
+  are both included in 2022 holidays. If remove_holidays=['new year'], then after this line of code:
+      local_holidays.pop_named(record) 
+  the results will erroneously include 2021's holidays as well. But practically no one would delete New Year anyway, we should be fine and there is no need to alarm the user about this issue.
+"""
+def non_business_days(date, interval=0, country='US', state='MA', province=None, add_holidays=None, remove_holidays=None, first_n_dates=0) -> dict:      
+  # 1. Collect weekends, standard holidays and user defined add/remove holidays
+  # 1.1 Get all saturdays and sundays in the given year  
+  date = as_datetime(date)
+  year = date.year
```

Comment:
How do you see this being used on year boundaries? I see this API mostly being used to get which days courts are available in the next 2-3 months, so it might make sense if the date is in November or December to also generate the dates in the next year as well.

Reply:
I'm realizing that the holidays to add also contain the year, so that might make it more difficult to generate holidays into the next year (and in general, increase the maintenance cost for keeping hardcoded holidays up-to-date as the years go on). Maybe accepting dates not as strings but as general year-independent rules (like the `holidays` [package does internally](https://github.com/dr-prodigy/python-holidays/blob/7ced8bf6235b17fbe57b69304c4537426653639e/holidays/countries/united_states.py#L178)) would make sense.

Reply:
Good comments! 

Waiting for a future version update sounds good to me, as we may also collect some feedback from users by then. I'll leave this conversation open so we can revisit when the time comes.